--- a/net/minecraft/client/network/NetHandlerPlayClient.java
+++ b/net/minecraft/client/network/NetHandlerPlayClient.java
@@ -1431,7 +1431,8 @@
       p_175093_1_.func_179788_a(this.field_147300_g.func_175723_af());
    }
 
+   @SuppressWarnings("incomplete-switch")
    @Override
    public void func_175099_a(SPacketTitle p_175099_1_) {
       PacketThreadUtil.func_180031_a(p_175099_1_, this, this.field_147299_f);
       SPacketTitle.Type spackettitle$type = p_175099_1_.func_179807_a();
@@ -1478,7 +1479,8 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    @Override
    public void func_147256_a(SPacketPlayerListItem p_147256_1_) {
       PacketThreadUtil.func_180031_a(p_147256_1_, this, this.field_147299_f);
 
