--- a/net/minecraft/nbt/JsonToNBT.java
+++ b/net/minecraft/nbt/JsonToNBT.java
@@ -260,7 +260,7 @@
          if (p_193603_2_ == 1) {
-            list.add((T)((NBTPrimitive)nbtbase).func_150290_f());
+            list.add((T)(Byte)((NBTPrimitive)nbtbase).func_150290_f());
          } else if (p_193603_2_ == 4) {
-            list.add((T)((NBTPrimitive)nbtbase).func_150291_c());
+            list.add((T)(Long)((NBTPrimitive)nbtbase).func_150291_c());
          } else {
-            list.add((T)((NBTPrimitive)nbtbase).func_150287_d());
+            list.add((T)(Integer)((NBTPrimitive)nbtbase).func_150287_d());
          }
