--- a/net/minecraft/client/renderer/tileentity/TileEntityRendererDispatcher.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntityRendererDispatcher.java
@@ -77,7 +77,7 @@
          this.field_147559_m.put(p_147546_1_, tileentityspecialrenderer);
       }
 
-      return tileentityspecialrenderer;
+      return (TileEntitySpecialRenderer<T>) tileentityspecialrenderer;
    }
 
    @Nullable