--- a/net/minecraft/block/BlockHugeMushroom.java
+++ b/net/minecraft/block/BlockHugeMushroom.java
@@ -156,7 +156,8 @@
             return p_185499_1_;
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    @Override
    public IBlockState func_185471_a(IBlockState p_185471_1_, Mirror p_185471_2_) {
       BlockHugeMushroom.EnumType blockhugemushroom$enumtype = p_185471_1_.func_177229_b(field_176380_a);
       switch(p_185471_2_) {
