--- a/net/minecraft/advancements/AdvancementList.java
+++ b/net/minecraft/advancements/AdvancementList.java
@@ -20,7 +20,7 @@
    private AdvancementList.Listener field_192095_e;
 
    public void func_192083_a(Map<ResourceLocation, Advancement.Builder> p_192083_1_) {
-      Function<ResourceLocation, Advancement> function = Functions.forMap(this.field_192092_b, (Advancement)null);
+      Function<ResourceLocation, Advancement> function = Functions.forMap(this.field_192092_b, null);

       while(!p_192083_1_.isEmpty()) {
