--- a/net/minecraft/command/EntitySelector.java
+++ b/net/minecraft/command/EntitySelector.java
@@ -270,7 +270,7 @@
 
    private static List<Predicate<Entity>> func_179659_d(Map<String, String> p_179659_0_) {
       List<Predicate<Entity>> list = Lists.newArrayList();
-      final String s = func_179651_b(p_179659_0_, field_190847_u);
+      String s = func_179651_b(p_179659_0_, field_190847_u);
       final boolean flag = s != null && s.startsWith("!");
       if (flag) {
          s = s.substring(1);
