--- a/net/minecraft/block/BlockFurnace.java
+++ b/net/minecraft/block/BlockFurnace.java
@@ -69,8 +69,9 @@
          p_176445_1_.func_180501_a(p_176445_2_, p_176445_3_.func_177226_a(field_176447_a, enumfacing), 2);
       }
    }

+   @SuppressWarnings("incomplete-switch")
    @SideOnly(Side.CLIENT)
    @Override
    public void func_180655_c(IBlockState p_180655_1_, World p_180655_2_, BlockPos p_180655_3_, Random p_180655_4_) {
       if (this.field_149932_b) {