--- a/net/minecraft/client/gui/GuiOverlayDebug.java
+++ b/net/minecraft/client/gui/GuiOverlayDebug.java
@@ -87,6 +87,7 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    protected List<String> call() {
       BlockPos blockpos = new BlockPos(
          this.field_175242_a.func_175606_aa().field_70165_t,
@@ -240,9 +241,9 @@
             list.add("");
             list.add(String.valueOf(Block.field_149771_c.func_177774_c(iblockstate.func_177230_c())));
 
-            for(Entry<IProperty<?>, Comparable<?>> entry : iblockstate.func_177228_b().entrySet()) {
+            for(Entry entry : iblockstate.func_177228_b().entrySet()) {
                IProperty<T> iproperty = (IProperty)entry.getKey();
                T t = (T)entry.getValue();
                String s = iproperty.func_177702_a(t);
                if (Boolean.TRUE.equals(t)) {
                   s = TextFormatting.GREEN + s;
