--- a/net/minecraft/util/JsonUtils.java
+++ b/net/minecraft/util/JsonUtils.java
@@ -261,6 +261,6 @@
    public static <T> T func_193838_a(Gson p_193838_0_, Reader p_193838_1_, Type p_193838_2_, boolean p_193838_3_) {
       try {
          JsonReader jsonreader = new JsonReader(p_193838_1_);
          jsonreader.setLenient(p_193838_3_);
-         return p_193838_0_.getAdapter(TypeToken.get(p_193838_2_)).read(jsonreader);
+         return (T) p_193838_0_.getAdapter(TypeToken.get(p_193838_2_)).read(jsonreader);
       } catch (IOException ioexception) {
