--- a/net/minecraft/block/BlockRailPowered.java
+++ b/net/minecraft/block/BlockRailPowered.java
@@ -34,6 +34,7 @@
       );
    }
 
+   @SuppressWarnings("incomplete-switch")
    protected boolean func_176566_a(World p_176566_1_, BlockPos p_176566_2_, IBlockState p_176566_3_, boolean p_176566_4_, int p_176566_5_) {
       if (p_176566_5_ >= 8) {
          return false;
@@ -179,7 +180,8 @@
       return i;
    }
 
+   @SuppressWarnings("incomplete-switch")
    @Override
    public IBlockState func_185499_a(IBlockState p_185499_1_, Rotation p_185499_2_) {
       switch(p_185499_2_) {
          case CLOCKWISE_180:
@@ -252,7 +254,8 @@
       }
    }
 
+   @SuppressWarnings("incomplete-switch")
    @Override
    public IBlockState func_185471_a(IBlockState p_185471_1_, Mirror p_185471_2_) {
       BlockRailBase.EnumRailDirection blockrailbase$enumraildirection = p_185471_1_.func_177229_b(field_176568_b);
       switch(p_185471_2_) {
