--- a/net/minecraft/network/NetworkManager.java
+++ b/net/minecraft/network/NetworkManager.java
@@ -115,7 +115,7 @@
       this.func_150718_a(textcomponenttranslation);
    }
 
-   protected void channelRead0(ChannelHandlerContext p_channelRead0_1_, Packet<?> p_channelRead0_2_) throws Exception {
+   protected void channelRead0(ChannelHandlerContext p_channelRead0_1_, Packet p_channelRead0_2_) throws Exception {
       if (this.field_150746_k.isOpen()) {
          try {
             p_channelRead0_2_.func_148833_a(this.field_150744_m);
