--- a/net/minecraft/world/gen/structure/StructureStrongholdPieces.java
+++ b/net/minecraft/world/gen/structure/StructureStrongholdPieces.java
@@ -26,6 +26,7 @@
 import net.minecraft.world.gen.structure.template.TemplateManager;
 import net.minecraft.world.storage.loot.LootTableList;
 
+@SuppressWarnings("incomplete-switch")
 public class StructureStrongholdPieces {
    private static final StructureStrongholdPieces.PieceWeight[] field_75205_b = new StructureStrongholdPieces.PieceWeight[]{
       new StructureStrongholdPieces.PieceWeight(StructureStrongholdPieces.Straight.class, 40, 0),
