--- a/net/minecraft/entity/ai/EntityAINearestAttackableTarget.java
+++ b/net/minecraft/entity/ai/EntityAINearestAttackableTarget.java
@@ -67,7 +67,7 @@
             return true;
          }
       } else {
-         this.field_75309_a = this.field_75299_d
+         this.field_75309_a = (T)this.field_75299_d
             .field_70170_p
             .func_184150_a(
                this.field_75299_d.field_70165_t,
@@ -99,9 +99,9 @@

                      return 1.0;
                   }
                },
-               this.field_82643_g
+               (Predicate<EntityPlayer>)this.field_82643_g);
             );
          return this.field_75309_a != null;
       }
    }